### ISSUE TITLE
Consolidating differences between my warp-ctc and espnet

### ISSUE THREAD

**Issue by SeanNaren** [espnet/espnet#1215]
Hey guys! Thanks so much for the work you've guys done, this library is looking awesome!

I noticed that you guys have your own warp-ctc fork and added pypi support to it which is really slick. What other changes have you guys made or plan to make?

One thing I'm curious about is the possibility of deprecating my warp-ctc repo, and using your support pypi repo as the default, would that be alright with you guys?

**Comment by sw005320** [comment]
> Hey guys! Thanks so much for the work you've guys done, this library is looking awesome!

@SeanNaren, first thanks a lot for your great work of the warp CTC pytorch implementation. We cannot make our system without your great work.
We really appreciate it.

> I noticed that you guys have your own warp-ctc fork and added pypi support to it which is really slick. What other changes have you guys made or plan to make?

We will not make significant changes for it, and we just try to maintain as it is with some bug fix if there are some or small interface extensions for our use if needed.

> One thing I'm curious about is the possibility of deprecating my warp-ctc repo, and using your support pypi repo as the default, would that be alright with you guys?

I personally think this is a good idea since we'll keep this maintenance anyway for our use and this effort can be contributed not only to ESPnet users but other people that want to simply use your warp-ctc. I just want to ask the opinion from @jnishi and @ysk24ok, who are actually working on it.

**Comment by ysk24ok** [comment]
I also agree with the idea of integrating those diverged repositories.

> using your support pypi repo as the default

I don't know how to do this specifically. espnet/warp-ctc is forked from SeanNaren/warp-ctc and users think the latter is the default.
Transferring ownership of SeanNaren/warp-ctc to espnet org written in  https://help.github.com/en/enterprise/2.16/user/articles/transferring-a-repository? But the page says `The target account must not have a repository with the same name, or a fork in the same network.`. espnet org already have warp-ctc repo, so transferring ownership seems difficult.

**Comment by SeanNaren** [comment]
Oh I meant I'll turn it to read-only and make a change in the README to look at the espnet supported repo rather than transfer ownership. Might cause more confusion than anything!

**Comment by sw005320** [comment]
> Oh I meant I'll turn it to read-only and make a change in the README to look at the espnet supported repo rather than transfer ownership. Might cause more confusion than anything!

I think this is an easy solution. If @ysk24ok is OK, let's do this.


> On a separate note what do you guys think about pytorch's ctc loss function? Have you thought about migrating to it?

We had some investigation on the old PyTorch (1.1) CTC and found some degradation compared with warp CTC. But this story can be changed for PyTorch 1.2 since I observer several code-level refinements for CTC on that version. Right now, we would not fully move to PyTorch 1.2 and cannot make such comparisons, but we'll think of it then.

**Comment by ysk24ok** [comment]
> I'll turn it to read-only and make a change in the README to look at the espnet supported repo rather than transfer ownership.

This sounds easier and simpler 👍

**Comment by SeanNaren** [comment]
> We had some investigation on the old PyTorch (1.1) CTC and found some degradation compared with warp CTC. But this story can be changed for PyTorch 1.2 since I observer several code-level refinements for CTC on that version. Right now, we would not fully move to PyTorch 1.2 and cannot make such comparisons, but we'll think of it then.

Sounds good, if we make any headway on that internally I'll let you guys know! 

Also on the ESPnet warp-ctc it says you guys don't support Pytorch 1.2 (https://github.com/espnet/warp-ctc#for-pytorch-12). Any reason behind this? We've been running fine using warp-ctc with Pytorch 1.2

**Comment by ysk24ok** [comment]
It just means currently warp-ctc is not tested with PyTorch 1.2 on TravisCI and wheels for that version are not provided. We can support PyTorch 1.2 by testing and providing wheels, if there is no difficulty.
@sw005320 @jnishi what do you think?

**Comment by sw005320** [comment]
> Also on the ESPnet warp-ctc it says you guys don't support Pytorch 1.2 (https://github.com/espnet/warp-ctc#for-pytorch-12). Any reason behind this? We've been running fine using warp-ctc with Pytorch 1.2

Simply because main ESPnet does not support PyTorch 1.2 (e.g., https://github.com/espnet/espnet/issues/1194)

> It just means currently warp-ctc is not tested with PyTorch 1.2 on TravisCI and wheels for that version are not provided. We can support PyTorch 1.2 by testing and providing wheels, if there is no difficulty.
@sw005320 @jnishi what do you think?

Yes, let's do it.

**Comment by jnishi** [comment]
> We can support PyTorch 1.2 by testing and providing wheels, if there is no difficulty.

It sounds good!

However, is there any way to check whether warp-ctc work fine with Pytorch 1.2?
If there isn't, it would be better to add test code to this repository. For example, it will be good to check if warp-ctc's loss is same as  pytorch's loss.  cf. https://github.com/espnet/espnet/issues/700#issuecomment-520149874.
